python/dataset_plotting.py

```diff
@@ -0,0 +1,286 @@
+# variable are returned.
+def plot_image(input_data, contours=False, plot=True):
+
+    ndim = len(input_data.dimensions())
+    # TODO: this currently allows for plot_image to be called with a 3D dataset
+    # and plot=False, which would lead to an error. We should think of a better
+    # way to protect against this.
+    if (ndim > 1) and ((ndim < 3) or ((ndim < 5) and not plot)):
+
+        values = []
+        for var in input_data:
+            if var.is_data:
+                values.append(var)
+
+        if len(values) > 1:
+            raise RuntimeError("More than one Data.Value found! Please use e.g."
+                               " plot_image(dataset.subset(Data.Value, 'sample'))"
+                               " to select only a single Value.")
+
+        xcoord = input_data[dimensionCoord(values[0].dimensions.labels[0])]
+        ycoord = input_data[dimensionCoord(values[0].dimensions.labels[1])]
+        x = xcoord.numpy
+        y = ycoord.numpy
+        xmin = np.amin(x)
+        xmax = np.amax(x)
+        ymin = np.amin(y)
+        ymax = np.amax(y)
+
+        ratio = (ymax - ymin) / (xmax - xmin)
+
+        layout = dict(
+            autosize=False,
+            width=800,
+            height=800*ratio,
+            xaxis = dict(
+                    range = [xmin,xmax],
+                    title = "{} [{}]".format(xcoord.name, xcoord.unit)),
+            yaxis = dict(
+                    range = [ymin,ymax],
+                    title = "{} [{}]".format(ycoord.name, ycoord.unit))
+            )
+
+        if plot:
+            if contours:
+                data = [go.Contour(
+                    x = x,
+                    y = y,
+                    z = values[0].numpy,
+                    colorscale = 'Viridis',
+                    colorbar=dict(
+                        title="{} [{}]".format(values[0].name,values[0].unit),
+                        titleside = 'right',
+                        )
+                    )]
+            else:
+                xmin -= 0.5*(x[1]-x[0])
+                xmax += 0.5*(x[-1]-x[-2])
+                ymin -= 0.5*(y[1]-y[0])
+                ymax += 0.5*(y[-1]-y[-2])
+                
+                data = [go.Heatmap(
+                    z = values[0].numpy,
+                    colorscale = 'Viridis',
+                    colorbar=dict(
+                        title="{} [{}]".format(values[0].name,values[0].unit),
+                        titleside = 'right',
+                        )
+                    )]
+            return iplot(dict(data=data, layout=layout))
+        else:
+            return [values[0], layout]
+
+    else:
+        raise RuntimeError("Unsupported number of dimensions in plot_image.")
+
+#===============================================================================
+
+# Plot a 2D slice through a 3D dataset with a slider to adjust the position of
+# the slice in the third dimension.
+def plot_sliceviewer(input_data):
```

Comment:
Akin to the MantidPlot sliceviewer and datasetviewer, we would eventually want to dynamically change what dimensions were mapped to plot x and ploty. Any thoughts in this direction?

Reply:
i think this would be more something for the `datasetviewer`? But we could support it by using something like `plot_sliceviewer(d, xy_axes=[Data.X, Data.Z])`